src/System.Collections.Specialized/src/System/Collections/Specialized/NameObjectCollectionBase.cs

```diff
@@ -24,11 +24,11 @@ namespace System.Collections.Specialized
     public abstract class NameObjectCollectionBase : ICollection, ISerializable, IDeserializationCallback
     {
         private bool _readOnly = false;
-        private ArrayList _entriesArray;
+        private ArrayList _entriesArray = null!;
```

Comment:
Can you add a short comment about why the ! is ok here? e.g. "// initialized in Reset method called from ctor".  Same for _entriesTable below.

Reply:
sure